launcher/res/qml/ActionListing.qml

```diff
@@ -23,6 +23,17 @@ Flow {
 
             contentItem: ColumnLayout {
 
+                ImageAction {
```

Comment:
Having a closer look at this, when you say you want both an icon and an image, it looks here like you want them both visible, on the same action, at the same time? Rather than each icon having either an icon or an image, with the Launcher being able to display actions with either at the same time?

Could you post a screenshot of what you have in mind? That would help clear things up I think.

Reply:
If you want just one visible at the same time, based on whether it's an image or icon, you could potentially use the [Loader](https://doc.qt.io/qt-5/qml-qtquick-loader.html) entity, to have an if-statement on the `url` part. The if-statement could identify whether it `endswith(".png")` or not, to determine whether to pass `AwesomeIcon`, or `ImageAction`.

Also, could we rename `ImageAction` to something like `ImageIcon`? It doesn't appear to be an actual action, but rather an icon coming from an image, rather than AwesomeFont.

Reply:
> it looks here like you want them both visible

No, that's something I want to avoid because they are at same spot. Sorry for my explanatory skills. I wasn't able to have action with awesome and another action with png at the same time with different approach.

> you could potentially use the Loader entity

I have lack of QML knowledge I even don't know how to start so if you have in mind how to do it, do it please. If I understood correctly you can do *if* statement inside QML which decide if will look into awesome list or use it as path (with appending to `AVALON_ICON_ROOT` env value) so `res_path` can be removed from engine and icons from launcher.

> Also, could we rename ImageAction to something like ImageIcon?

Sure. Will this entity be necessary with the `Loader`?

Reply:
> Sure. Will this entity be necessary with the Loader?

Yeah, the Loader is basically a way to return different items, based on some expression. It's like a gateway, taking your expression into account, and then turning into the item you return.

> I have lack of QML knowledge I even don't know how to start so if you have in mind how to do it, do it please.

Oh no! I won't have time for the next few weeks, but I'd be happy to point you in the right direction. It should only be a few lines of code.

Here's an example you could run standalone, using the `$QTDIR/bin/qmlscene.exe`; it's like the Python interpreter of QML.

- https://github.com/pyblish/pyblish-qml/blob/8ff555e78fc5c9727c75d7f2182cbd0bdfc1b267/pyblish_qml/qml/tests/tst_ConditionalDelegate.qml#L40

Notice how it outputs different items based on that expression? What you want is to output `AwesomeIcon`, unless the `icon` ends with `.png`, in which case you output your `ImageAction`

Reply:
Ok, right now I'm busy too but I'll try it as soon as possible. Should I close this pull request or let it open?

Reply:
You could keep it open and push the potential fix into this branch, right?

Reply:
Happy to leave it open as a reminder, anyone is welcome to try and tackle this and contributors are able to push directly to your branch as well.